paasta_tools/setup_tron_namespace.py

```diff
@@ -62,22 +75,49 @@ def main():
     else:
         logging.basicConfig(level=logging.WARNING)
 
-    try:
-        new_config = tron_tools.create_complete_config(
-            service=args.service,
-            soa_dir=args.soa_dir,
-        )
+        if args.all_namepsaces:
+            tron_config = tron_tools.load_tron_config()
+            try:
+                services = tron_tools.get_tron_namespaces_for_cluster(
+                    cluster=tron_config.get_tron_cluster())
+            except ConflictingNamespacesError as e:
+                log.error('Failed to list tron namespaces: {error}'.format(
+                    error=str(e)))
+                sys.exit(1)
+        else:
+            services = [args.service]
+
         client = tron_tools.get_tron_client()
-        client.update_namespace(args.service, new_config)
-    except (
-        InvalidTronConfig,
-        NoConfigurationForServiceError,
-        TronNotConfigured,
-        PaastaNotConfiguredError,
-        TronRequestError,
-    ) as e:
-        log.error('Update for {namespace} failed: {error}'.format(namespace=args.service, error=str(e)))
-        sys.exit(1)
+
+        updated = []
+        failed = []
+
+        for service in services:
+            try:
+                new_config = tron_tools.create_complete_config(
+                    service=args.service,
+                    soa_dir=args.soa_dir,
+                )
+                client.update_namespace(args.service, new_config)
+                updated.append(service)
+            except (
+                    InvalidTronConfig,
```

Comment:
Are they all the errors that might happen here? If not, maybe we can catch a general error here. I can't come up with the scenario that we should raise the exception rather than continue to work on other services.

Reply:
i had the same question. in the end i just moved that catch block as is. if we swallow all exceptions here we might miss a real issue that needs to be addressed. i'll leave it for now.